design/pipeline-triggers.md

```diff
@@ -0,0 +1,369 @@
+# Triggers in pipelines
+
+Triggers are events on which you can start your pipeline run automatically. You can enable triggers on your pipeline by subscribing to both internal and external events.
+
+At high level there are 3 different types of pipeline triggers.
+1. Resource triggers
+2. Schedule triggers
+3. Webhook triggers
+
+## Resource triggers
+You can enable triggers on the resources defined in your pipeline. Resources can be of types pipelines, repositories, containers and packages.
+Triggers are enabled by default on all the resources. However, you can choose to override/disable triggers for each resource.
+
+
+
+### Pipelines
+A new pipeline is triggered automatically whenever a new run of the `pipeline` resource is succesfully completed. See [pipeline resources](pipeline-resources.md#resources-pipelines) for more details.
+
+#### Scenarios
+- I would like to trigger my pipeline when a new ‘helm-package’ artifact is published by ‘Helm-CI’ pipeline that ran on `releases/*` branch.
```

Comment:
This suggests you want to filter on produced artifact names, but I don't see anything like that below.

Reply:
If we look at CD scenarios, usually, the artifact name doesn't change from run to run. However what changes from run to run can be branch or tag etc.

Reply:
OK, that makes sense. Might want to update the scenario, then, to make it unambiguous what you're trying to support.

Reply:
Updated the scenario accordingly.